ext/odbc/odbc.stub.php

```diff
@@ -4,193 +4,193 @@
 
 function odbc_close_all(): void {}
 
-/** @param resource $result_id */
-function odbc_binmode($result_id, int $mode): bool {}
+/** @param resource $stmt */
+function odbc_binmode($stmt, int $mode): bool {}
 
-/** @param resource $result_id */
-function odbc_longreadlen($result_id, int $length): bool {}
+/** @param resource $stmt */
+function odbc_longreadlen($stmt, int $length): bool {}
 
 /**
- * @param resource $connection_id
+ * @param resource $odbc
  * @return resource|false
  */
-function odbc_prepare($connection_id, string $query) {}
+function odbc_prepare($odbc, string $query) {}
 
-/** @param resource $result_id */
-function odbc_execute($result_id, array $parameters_array = []): bool {}
+/** @param resource $stmt */
+function odbc_execute($stmt, array $params = []): bool {}
 
-/** @param resource $result_id */
-function odbc_cursor($result_id): string|false {}
+/** @param resource $stmt */
+function odbc_cursor($stmt): string|false {}
 
 #ifdef HAVE_SQLDATASOURCES
-/** @param resource $connection_id */
-function odbc_data_source($connection_id, int $fetch_type): array|false {}
+/** @param resource $odbc */
+function odbc_data_source($odbc, int $fetch_type): array|false {}
 #endif
 
 /**
- * @param resource $connection_id
+ * @param resource $odbc
  * @return resource|false
  */
-function odbc_exec($connection_id, string $query) {}
+function odbc_exec($odbc, string $query) {}
 
 /**
- * @param resource $connection_id
+ * @param resource $odbc
  * @return resource|false
  * @alias odbc_exec
  */
-function odbc_do($connection_id, string $query) {}
+function odbc_do($odbc, string $query) {}
 
 #ifdef PHP_ODBC_HAVE_FETCH_HASH
-/** @param resource $result */
-function odbc_fetch_object($result, int $rownumber = -1): stdClass|false {}
+/** @param resource $stmt */
+function odbc_fetch_object($stmt, int $row_number = -1): stdClass|false {}
```

Comment:
Over in #6294 I used just `$row` for this. If not that, we should use `$row_num`, as we consistently abbreviate "number".

Reply:
There is `odbc_fetch_into()` which fetches a row into a by-ref parameter (so it can return the number of columns ...), which is currently called `$row`. We could rename that param to `$array`, and then use `$row` instead of `$row_number`.

Reply:
Yes, I think it would make sense to rename that parameter to `$array`. If I understood right, odbc_fetch_into does exactly the same as odbc_fetch_array, just writing by reference instead of returning, right?

As a side-note, I'm wondering if we can drop support for ODBC flavors that do not HAVE_SQL_EXTENDED_FETCH. That seems to be "solid" and "empress".

Reply:
> If I understood right, odbc_fetch_into does exactly the same as odbc_fetch_array, just writing by reference instead of returning, right?

Besides this, another difference is that `odbc_fetch_into()` fetches an *indexed* array, while `odbc_fetch_array()` fetches an *associative* array. And since the code is duplicated (sigh), there may be small behavioral differences as well (doesn't seem so from a quick glance, though).

> As a side-note, I'm wondering if we can drop support for ODBC flavors that do not HAVE_SQL_EXTENDED_FETCH. That seems to be "solid" and "empress".

I *assume* that quite some old stuff could be dropped. Do we still really need to support ODBC < 3.0? Anyway, it seems to me that [SolidDB supports `SQLExtendedFetch()`](https://support.unicomsi.com/manuals/soliddb/100/Programmer_Guide/wwhelp/wwhimpl/common/html/wwhelp.htm#href=2_ODBC_API.04.29.html&single=true) (maybe there are old driver versions still in use which don't, though). This would need more investigation (or maybe just a bold move to drop it in "master", waiting for feedback).